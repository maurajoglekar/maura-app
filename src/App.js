import logo from "./logo.svg";
import "./App.css";
import { useState, useEffect } from "react";
import styled from "styled-components";

// get products and navigate to product details using router
// use API: https://fakestoreapi.com/products
// write unit tests

const StyledProducts = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  justify-content: center;
  padding: 24px;
  background-color: lightgray;
`;

const StyledCard = styled.div`
  height: 300px;
  width: 250px;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  overflow: hidden;
  padding: 16px 8px;
  text-align: center;
  background-color: white;

  img {
    height: 10rem;
    width: 10rem;
  }

  .product-title {
    height: 50px;
  }
`;

function App() {
  const [products, setProducts] = useState([]);

  const fetchProducts = async () => {
    try {
      const response = await fetch("https://fakestoreapi.com/products");
      const data = await response.json();
      setProducts(data);
    } catch (error) {
      console.log("Error when getting products: ${error}");
    }
  };

  useEffect(() => {
    fetchProducts();
  }, []);
  return (
    <StyledProducts>
      {products.map((product) => {
        return (
          <StyledCard key={product.id}>
            <img src={product.image} alt={product.title} />
            <div className="product-title">{product.title}</div>
          </StyledCard>
        );
      })}
    </StyledProducts>
  );
}

export default App;
