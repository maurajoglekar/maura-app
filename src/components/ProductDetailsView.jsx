import styled from "styled-components";

const StyledProductContainer = styled.div`
  display: flex;
  flex-wrap: nowrap;
  justify-content: center;
`;

const StyledProduct = styled.div`
  display: flex;
  gap: 40px;
  margin-top: 24px;

  img {
    width: 500px;
    height: 600px;
    border-radius: 12px;
    padding: 20px;
    background-color: white;
  }
`;

const StyledDetails = styled.div`
  display: flex;
  justify-content: start;
  flex-direction: column;
  width: 600px;
  align-items: start;
  gap: 30px;
  text-align: start;
`;

const ProductDetailsView = ({ details }) => {
  return (
    <StyledProductContainer>
      {details && (
        <StyledProduct>
          <img src={details.image} alt={details.title} />
          <StyledDetails>
            <h3 data-testid="title">{details.title}</h3>
            <div>{details.description}</div>
            <div>Price: ${details.price}</div>
            <div>Customer Rating: {details.rating.rate}</div>
          </StyledDetails>
        </StyledProduct>
      )}
    </StyledProductContainer>
  );
};

export default ProductDetailsView;
