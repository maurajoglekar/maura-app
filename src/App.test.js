import { render } from "@testing-library/react";
import App from "./App";
import ProductDetailsView from "./components/ProductDetailsView";

const title = "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops";
const mockedProducts = [
  {
    id: 1,
    title: title,
    price: 109.95,
    description:
      "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
    category: "men's clothing",
    image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
    rating: {
      rate: 3.9,
      count: 120,
    },
  },
];
test("renders store header", () => {
  const elem = render(<App />);
  const headerElem = elem.getByText(/ecommerce site/i);
  expect(headerElem).toBeInTheDocument();
});

test("renders product details", () => {
  const elem = render(<ProductDetailsView details={mockedProducts[0]} />);
  expect(elem).toMatchSnapshot();
});

test("renders product details title", () => {
  const elem = render(<ProductDetailsView details={mockedProducts[0]} />);
  const titleElem = elem.getByTestId("title");
  expect(titleElem).toBeInTheDocument();
});
