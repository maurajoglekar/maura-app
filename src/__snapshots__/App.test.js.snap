// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders product details 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ghWlax jRjUoo"
      >
        <div
          class="sc-kLhKbu gxxUYy"
        >
          <img
            alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
          />
          <div
            class="sc-ixGGxD ekIrVL"
          >
            <h3
              data-testid="title"
            >
              Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
            </h3>
            <div>
              Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
            </div>
            <div>
              Price: $
              109.95
            </div>
            <div>
              Customer Rating: 
              3.9
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ghWlax jRjUoo"
    >
      <div
        class="sc-kLhKbu gxxUYy"
      >
        <img
          alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
          src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
        />
        <div
          class="sc-ixGGxD ekIrVL"
        >
          <h3
            data-testid="title"
          >
            Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
          </h3>
          <div>
            Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
          </div>
          <div>
            Price: $
            109.95
          </div>
          <div>
            Customer Rating: 
            3.9
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
